examples/Browser/Server/Startup.cs

```diff
@@ -42,17 +55,42 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                 app.UseHsts();
             }
 
-            app.UseHttpsRedirection();
+            // app.UseHttpsRedirection();
             app.UseDefaultFiles();
             app.UseStaticFiles();
-
+            app.UseSpaStaticFiles();
             app.UseRouting();
+            app.UseCors();
 
             app.UseGrpcWeb();
 
             app.UseEndpoints(endpoints =>
             {
-                endpoints.MapGrpcService<GreeterService>().EnableGrpcWeb();
+                endpoints.MapGrpcService<GreeterService>()
+                .RequireCors("AllowAll")
+                .EnableGrpcWeb();
+            });
+
+            app.UseSpa(spa =>
+            {
+                if (env.IsDevelopment())
+                {
+                    var fileName = "npm";
+                    var args = "run build:watch";
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    {
+                        fileName = "cmd";
+                        args = $"/c npm {args}";
+                    }
+
+                    // Why is NpmRunner not public???
+                    Process.Start(new ProcessStartInfo
```

Comment:
Is this usual for SPA apps? Running npm on app startup instead of build is something I have never seen before.

Reply:
@JamesNK aspnet core official extension for Angular and React do this with internal NpmRunner, this code here is actually copied from there since NpmRunner is not public : (